### ISSUE TITLE
Speaker Classification

### ISSUE THREAD

**Issue by JacksonMaxfield** [CouncilDataProject/cdp-roadmap#5]
## Use Case
_Please provide a use case to help us understand your request in context_
There has been a long term goal (wish) that we could provide speaker annotations next to each text block of the transcripts. This provides simply a better user experience and produces a richer dataset.

## Solution
_Please describe your ideal solution_
Early thinking is a voting classifier using a mix of random forests, naive bayes, etc. trained on a dataset of annotated text blocks.

## Alternatives
_Please describe any alternatives you've considered, even if you've dismissed them_
Use Google's and other Cloud providers implementations of what seems to be NN cluster for speaker annotations but very costly and has a limit to n speakers.

## Stakeholders
_Please add any individual person or team's that should be brought in for discussion on the project_
Backend, Frontent, UX

## Major Components
_Please add any major components that need to be done for this project_
* [ ] Construct data cleaning pipeline
* [ ] Annotate dataset
* [ ] Experiment with models
* [ ] CI/CD to always update the model used by CDP production deployment with best model

## Dependencies
_Please add any other major or minor project dependencies here_

## Other Notes
_Please add any extra notes here_
See cdptools/speakerbox for early experiments.

**Comment by JacksonMaxfield** [comment]
[prodigy](https://prodi.gy/) looks great for annotation: [manual audio annotation](https://prodi.gy/docs/audio-video#manual)

**Comment by isaacna** [comment]
Would the training data be clips of a "known speaker" so that the classification model can label them properly rather than just grouping unknown speakers together? 

Also, if we build this we should also make sure to backfill transcripts that have unknown speakers once we have more training data. I think gathering the training data would be mostly manual?

**Comment by JacksonMaxfield** [comment]
Yes! The clips would be of a known speaker. Any prediction that has a low confidence value should be labeled as "Unknown speaker"
